packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -0,0 +1,172 @@
+const { ERC20, ERC721 } = Action;
+const { DetailedERC20 } = ERC20;
+const { NFT } = ERC721;
+import GnosisSafe from "./GnosisSafe";
+import { local as localAddresses } from "../../tasit-contracts/3rd-parties/gnosis/addresses";
+
+import Account from "../../tasit-account/dist/index";
+
+const { GnosisSafe: GNOSIS_SAFE_ADDRESS } = localAddresses;
+const ERC20_ADDRESS = "0x37E1A58dD465D33263D00185D065Ee36DD34CDb4";
+const NFT_ADDRESS = "0x0E86f209729bf54763789CDBcA9E8b94f0FD5333";
+
+// 100 gwei
+const GAS_PRICE = bigNumberify(`${1e11}`);
+
+const { ZERO, ONE } = constants;
+
+describe("GnosisSafe", () => {
+  let gnosisSafe;
+  let tokensOwner;
+  let johnWallet;
+  let ephemeralWallet;
+  let erc20;
+  let nft;
+
+  before("", async () => {
+    tokensOwner = accounts[0];
+    johnWallet = accounts[9];
+
+    ephemeralWallet = Account.create();
+
+    // Contract deployment setup with john (accounts[9]) as the only owner
+    // To change that, edit the file "tasit-contract/3rd-parties/gnosis/scripts/2_deploy_contracts.js"
+    gnosisSafe = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+    erc20 = new DetailedERC20(ERC20_ADDRESS);
+
+    nft = new NFT(NFT_ADDRESS);
+  });
+
+  beforeEach("", async () => {
+    const etherBalance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${etherBalance}`).to.equal(`${ZERO}`);
+
+    const erc20Balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${erc20Balance}`).to.equal(`${ZERO}`);
+
+    gnosisSafe.removeWallet();
+    erc20.removeWallet();
+  });
+
+  afterEach("", async () => {
+    expect(erc20.subscribedEventNames()).to.be.empty;
+    expect(gnosisSafe.subscribedEventNames()).to.be.empty;
+    expect(provider._events).to.be.empty;
+  });
+
+  describe("test cases that needs ETH deposit to the wallet", async () => {
+    beforeEach("faucet", async () => {
+      await etherFaucet(provider, johnWallet, GNOSIS_SAFE_ADDRESS, ONE);
+      const balance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+      expect(`${balance}`).to.equal(`${ONE}`);
+    });
+
+    it("wallet owner should withdraw some ethers", async () => {
+      const signers = [johnWallet];
+      const { address: toAddress } = johnWallet;
+      const value = ONE;
+
+      gnosisSafe.setWallet(johnWallet);
+      const execTxAction = await gnosisSafe.transferEther(
+        signers,
+        toAddress,
+        value
+      );
+      await execTxAction.waitForNonceToUpdate();
+
+      const balance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+      expect(`${balance}`).to.equal(`${ZERO}`);
+    });
+  });
+
+  describe("test cases that needs ERC20 deposit to the wallet", async () => {
+    beforeEach("faucet", async () => {
+      await erc20Faucet(erc20, tokensOwner, GNOSIS_SAFE_ADDRESS, ONE);
+      const balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+      expect(`${balance}`).to.equal(`${ONE}`);
+    });
+
+    it("wallet owner should withdraw some ERC20 tokens", async () => {
```

Comment:
```suggestion
    it("wallet owner should be able to send some ERC20 tokens", async () => {
```

Reply:
@marcelomorgado Feel free to accept this change

Reply:
I think that is deprecated since we are using that comment now:
"contract account should send ERC20 tokens back to owner"

Reply:
Thanks. Resolving.